src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -256,6 +256,8 @@ def get_custom_clouds(cli_ctx):
     known_cloud_names = [c.name for c in KNOWN_CLOUDS]
     return [c for c in get_clouds(cli_ctx) if c.name not in known_cloud_names]
 
+def get_cloud_name(cli_ctx, cloud_name):
+    return next((x.name for x in get_clouds(cli_ctx) if x.name.lower() == cloud_name.lower()), cloud_name)
```

Comment:
Let's make this private so as not to expand the public interface of core. `_get_cloud_name`

Reply:
Updated